tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,398 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Python source file includes CIFAR10 utils for Keras model.
+
+The utilities in this file are used to build a model with native Keras.
+This module file will be used in Transform and generic Trainer.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from typing import List, Text
+import absl
+import tensorflow as tf
+import tensorflow_transform as tft
+
+import flatbuffers
+# pylint: disable=g-direct-tensorflow-import
+from tflite_support import metadata as _metadata
+from tflite_support import metadata_schema_py_generated as _metadata_fb
+# pylint: enable=g-direct-tensorflow-import
+
+from tfx.components.trainer.rewriting import converters
+from tfx.components.trainer.rewriting import rewriter
+from tfx.components.trainer.rewriting import rewriter_factory
+
+from tfx.components.trainer.executor import TrainerFnArgs
+
+# When training on the whole dataset use following constants instead.
+# This setting should give ~91% accuracy on the whole test set
+# _TRAIN_DATA_SIZE = 50000
+# _EVAL_DATA_SIZE = 10000
+# _TRAIN_BATCH_SIZE = 64
+# _EVAL_BATCH_SIZE = 64
+# _CLASSIFIER_LEARNING_RATE = 3e-4
+# _FINETUNE_LEARNING_RATE = 5e-5
+# _CLASSIFIER_EPOCHS = 12
+
+_TRAIN_DATA_SIZE = 128
+_EVAL_DATA_SIZE = 128
+_TRAIN_BATCH_SIZE = 32
+_EVAL_BATCH_SIZE = 32
+_CLASSIFIER_LEARNING_RATE = 1e-3
+_FINETUNE_LEARNING_RATE = 7e-6
+_CLASSIFIER_EPOCHS = 30
+
+_IMAGE_KEY = 'image'
+_LABEL_KEY = 'label'
+
+_TFLITE_MODEL_NAME = 'tflite'
+_LABEL_MAP_FILE_PATH = 'cifar10/data/labels.txt'
```

Comment:
why not just a list here?

Reply:
We will need to add the label file as part of the TFLite metadata to the trained TFLite model in order to serve it in MLKit. That process (the _write_metadata function) requires an actual file instead of a list.